flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -217,4 +217,14 @@
 	 * @return self as ByteBuf implementation.
 	 */
 	ByteBuf asByteBuf();
+
+	/**
+	 * @return whether the buffer is compressed or not.
+	 */
+	boolean isCompressed();
+
+	/**
+	 * Tags the buffer as compressed or uncompressed.
+	 */
+	void setCompressed(boolean isCompressed);
```

Comment:
The compression property can be always got before constructing the Buffer instance I guess, so it is graceful to give that property in the constructor. So we can remove this interface method and make the internal variable as final.

Reply:
After reviewing the consumer side as I mentioned via https://github.com/apache/flink/pull/10375/commits/2f538ab20262fbe9981d4851b93e9bee04fc6ad0#r353537910,

It might be feasible to not touch `Buffer` class completely. On producer side we can judge whether to write compressed buffer via optional `BufferCompressor` variable in `ResultPartition. On consumer side we can also judge whether to decompress the buffer via optional `BufferDecompressor` variable in `SingleInputGate`. And both these optional compressor/decompressor are determined by global configuration of enabling compression or not.

Reply:
We need the compression tag to identify if the buffer is compressed at downstream for the downstream may receive a mixture of compressed and uncompressed buffer.